ledger/src/blockstore_db.rs

```diff
@@ -1415,39 +1425,48 @@ fn get_cf_options<C: 'static + Column + ColumnName>(
     options
 }
 
-/// Constructs and returns a ColumnFamilyDescriptor based on the
-/// specified ShredStorageType.
-fn new_cf_descriptor_shreds<C: 'static + Column + ColumnName>(
-    storage_type: &ShredStorageType,
+/// Creates and returns the column family descriptors for both data shreds and
+/// coding shreds column families.
+///
+/// @return a pair of ColumnFamilyDescriptor where the first / second elements
+/// are associated to the first / second template class respectively.
+fn new_cf_descriptor_pair_shreds<
+    D: 'static + Column + ColumnName, // Column Family for Data Shred
+    C: 'static + Column + ColumnName, // Column Family for Coding Shred
+>(
+    shred_storage_type: &ShredStorageType,
     access_type: &AccessType,
     oldest_slot: &OldestSlot,
-    max_cf_size: &u64,
-) -> ColumnFamilyDescriptor {
-    match storage_type {
-        ShredStorageType::RocksLevel => new_cf_descriptor::<C>(access_type, oldest_slot),
-        ShredStorageType::RocksFifo => {
-            if *max_cf_size > FIFO_WRITE_BUFFER_SIZE {
-                new_cf_descriptor_fifo::<C>(max_cf_size)
-            } else {
-                warn!(
-                    "{} cf_size must be greater than {} when using ShredStorageType::RocksFifo.",
-                    C::NAME,
-                    FIFO_WRITE_BUFFER_SIZE
-                );
-                warn!(
-                    "Fall back to ShredStorageType::RocksLevel for cf::{}.",
-                    C::NAME
-                );
-                new_cf_descriptor::<C>(access_type, oldest_slot)
+    fifo_options: &Option<BlockstoreRocksFifoOptions>,
+) -> (ColumnFamilyDescriptor, ColumnFamilyDescriptor) {
+    match shred_storage_type {
+        ShredStorageType::RocksLevel => (
+            new_cf_descriptor::<D>(access_type, oldest_slot),
+            new_cf_descriptor::<C>(access_type, oldest_slot),
+        ),
+        ShredStorageType::RocksFifo => match &fifo_options {
+            Some(fifo) => (
+                new_cf_descriptor_fifo::<D>(&fifo.shred_data_cf_size),
+                new_cf_descriptor_fifo::<C>(&fifo.shred_code_cf_size),
+            ),
+            None => {
+                panic!("RocksFifo is used without a valid BlockstoreRocksFifoOptions");
```

Comment:
I think this could be a good scenario to move `BlockstoreRocksFifoOptions` struct into `ShredStorageType::RocksFifo` enum case. The updated enum would then look like:
```
pub enum ShredStorageType {
    // Stores shreds under RocksDB's default compaction (level).
    RocksLevel,
    // (Experimental) Stores shreds under RocksDB's FIFO compaction which
    // allows ledger store to reclaim storage more efficiently with
    // lower I/O overhead.
    RocksFifo(BlockstoreRocksFifoOptions),
}
```
This snippet would then look like this
```
        ShredStorageType::RocksFifo(fifo_options) => (
            new_cf_descriptor_fifo::<D>(&fifo.shred_data_cf_size),
            new_cf_descriptor_fifo::<C>(&fifo.shred_code_cf_size),
        ),
```
which makes it impossible for the user to specify fifo without passing in fifo options.

This means we can drop`BlockstoreRocksFifoOptions` from `BlockstoreOptions` too; this might scale better if we start allowing more options / customization on individual columns

Reply:
Oh, nice!  Didn't know about this!  Will definitely do.  Neat!